### PR TITLE
Add possibility to cancel jobs by settings via API

### PR DESCRIPTION
**Pull request by garretraziel**
This adds possibility to cancel jobs by settings, for example to cancel all jobs for specified BUILD. This functionality was already provided by `job_cancel_by_settings` call, but it wasn't in API.

Note that although I tried to add tests, they fail - canceling running jobs in tests doesn't work for some reason (even canceling by `id`).

### CODE REVIEW

**Comment by garretraziel** [comment]
Note that canceling jobs in tests doesn't work even with original code. Perhaps I need to mock some function somewhere?

**Comment by aaannz** [comment]
I looked into it and it's as I thought. Running jobs are cancelled via workers.
When job_cancel is received by scheduler it enqueues `cancel` command to respective worker which then updates job status by `job_done` call.
On top of that, if I'm reading code right, running jobs state is not set to `cancelled` but `done` and result `incomplete`. Only `scheduled` jobs state is set to `cancelled`.

File: lib/OpenQA/WebAPI/Controller/API/V1/Job.pm [hunk 1]
```diff
@@ -300,13 +300,23 @@ sub restart {
     $self->render(json => {result => $res, test_url => \@urls});
 }
 
+# Used for both apiv1_cancel and apiv1_cancel_jobs
 sub cancel {
     my ($self) = @_;
-    my $jobid = int($self->param('jobid'));
+    my $jobid = $self->param('jobid');
```
Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/API/V1/Job.pm]: why did you remove `int()` from here and then added it everywhere else?

Comment (garretraziel) [inline lib/OpenQA/WebAPI/Controller/API/V1/Job.pm]: It showed "use of unitialized value in function int()" warning when called without job id.

Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/API/V1/Job.pm]: Ah, right. Thanks for clarification.
